src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,48 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    static final int DATA_NUMBER_IN_LINE = 0;
+    static final int NAME_NUMBER_IN_LINE = 1;
+    static final int HOURS_NUMBER_IN_LINE = 2;
+    static final int MONEY_PER_HOUR_NUMBER_IN_LINE = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate startDate;
+        LocalDate finishDate;
+        LocalDate carrentDate;
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        startDate = LocalDate.parse(dateFrom, formatter);
+        finishDate = LocalDate.parse(dateTo, formatter);
+        StringBuilder report = new StringBuilder("Report for period ")
+                .append(dateFrom)
+                .append(" - ").append(dateTo)
+                .append(System.lineSeparator());
+        String [] dataNames = new String[data.length];
+        String[] personData;
```

Comment:
can we work with a single `arrayOfData`?

Reply:
I created an additional array "dataNames" for easier name comparison and better readable code.

Reply:
we don't need this additional array, as well as third loop, two is enough